apple/RNGHVector.m

```diff
@@ -0,0 +1,67 @@
+//
+//  RNGHVector.m
+//  DoubleConversion
+//
+//  Created by Michał Bert on 05/08/2024.
+//
+
+#import "RNGHVector.h"
+#import <Foundation/Foundation.h>
+
+@implementation Vector
+
+- (id)initWithX:(double)x withY:(double)y
+{
+  if (self = [super init]) {
+    self.x = x;
+    self.y = y;
+
+    self.magnitude = hypot(x, y);
+
+    self.unitX = x / self.magnitude;
+    self.unitY = y / self.magnitude;
+  }
+
+  return self;
+}
+
++ (Vector *)fromDirection:(RNGestureHandlerDirection)direction
+{
+  switch (direction) {
+    case RNGestureHandlerDirectionRight:
+      return [[Vector alloc] initWithX:1 withY:0];
+    case RNGestureHandlerDirectionLeft:
+      return [[Vector alloc] initWithX:-1 withY:0];
+    case RNGestureHandlerDirectionUp:
+      return [[Vector alloc] initWithX:0 withY:1];
+    case RNGestureHandlerDirectionDown:
+      return [[Vector alloc] initWithX:0 withY:-1];
+    case RNGestureHandlerDirectionUpLeft:
+      return [[Vector alloc] initWithX:-1 withY:1];
+    case RNGestureHandlerDirectionUpRight:
+      return [[Vector alloc] initWithX:1 withY:1];
+    case RNGestureHandlerDirectionDownLeft:
+      return [[Vector alloc] initWithX:-1 withY:-1];
+    case RNGestureHandlerDirectionDownRight:
+      return [[Vector alloc] initWithX:1 withY:-1];
+    default:
+      return [[Vector alloc] initWithX:0 withY:0];
+  }
+}
```

Comment:
Would it be possible to store static vector objects and cloning them when needed instead of allocating new ones each time?

I believe that's how it's done on Android's native side.

Reply:
The thing is, this would require a lot of code and I'm not sure if performance will be much better. Do you think it will be worth to try @j-piasecki?

Reply:
Yeah, I'm not sure if it's worth the effort. If Vector was an object with heavy initialization then probably yes, but in this case it would add **a lot** of boilerplate (https://stackoverflow.com/a/40591961).